sandbox/services/patient-searcher.js

```diff
@@ -0,0 +1,150 @@
+const Boom = require('boom')
+const fs = require('fs')
+const lodash = require('lodash')
+
+const EXAMPLE_PATIENT_SMITH = JSON.parse(fs.readFileSync('mocks/Patient.json'))
+const EXAMPLE_PATIENT_SMYTHE = JSON.parse(fs.readFileSync('mocks/Patient-Jayne-Smyth.json'))
+
+let _request;
+
+module.exports.init = function(request) {
+    _request = request;
+    return exports
+}
+
+// Verify search contains parameters
+module.exports.requestContainsParameters = function() {
+    const searchMap = {
+        "_exact-match": true,
+        "_history": true,
+        "_fuzzy-match": true,
+        "_max-results": true,
+        family: '$.name[?(@.use="usual")].family', // Usual family name
+        given: true,
+        gender: true,
+        birthdate: true,
+        "death-date": true,
+        "address-postcode": true,
+        organisation: true,
+    };
+
+    let hasAnySearchParam = false
+    for (let p of Object.keys(searchMap)) {
+        if (_request.query[p]) {
+            hasAnySearchParam = true
+            break
+        }
+    }
+    return hasAnySearchParam
+}
+
+// Determine which 'search' to perform based on parameters passed
+module.exports.search = function() {
+
+    function containsSearchParameters(searchParameters) {
+
+        // Create new object that doesnt contain _max-result parameters as it value can change and is handled later
+        function removeMaxResultParameter(parameters) {
+            return lodash.pickBy(parameters, (value, key) => key !== "_max-results")
+        }
+
+        let searchParamsWithoutMaxResult = removeMaxResultParameter(searchParameters)
+        let queryParamsWithoutMaxResult = removeMaxResultParameter(_request.query)
+
+        // Verifies that search parameters match query parameters
+        if (!lodash.isEqual(searchParamsWithoutMaxResult, queryParamsWithoutMaxResult)) {
+            return false
+        }
+
+        return true
+    }
+    
+    function buildPatientResponse(examplePatients = [], searchScore = 1.0) {
```

Comment:
This function is getting a little busy - consider divorcing the matching part (where it decides which response to dispatch)  from the building part (where it renders a response)